tez-dag/src/main/java/org/apache/tez/client/LocalClient.java

```diff
@@ -403,7 +405,14 @@ protected DAGAppMaster createDAGAppMaster(ApplicationAttemptId applicationAttemp
   @Override
   public TezAppMasterStatus getAMStatus(Configuration configuration, ApplicationId appId,
       UserGroupInformation ugi) throws TezException, ServiceException, IOException {
-    return clientHandler.getTezAppMasterStatus();
+    if (isLocalWithoutNetwork) {
+      if (clientHandler == null) {
+        return TezAppMasterStatus.INITIALIZING;
+      }
+      return clientHandler.getTezAppMasterStatus();
+    } else {
```

Comment:
if - else can be simplified (no need for else.
Are we sure we need the super. getAMStatus call? I would assume that a FrameworkClient would call its own method right?

Reply:
else simplification - OK
yes we need it, this is called from:
https://github.com/apache/tez/blob/master/tez-api/src/main/java/org/apache/tez/client/TezClient.java#L834

if we don't want to use LocalClient's own logic for this codepath, we shall fallback to the parent's, which is FrameworkClient. getAMStatus (as LocalClient is a FrameworkClient)

Reply:
Oh I see, this part of the TEZ-4236 addition -- makes sense

Reply:
thanks!